src/backend/joanie/core/api/__init__.py

```diff
@@ -19,7 +21,9 @@ def exception_handler(exc, context):
             detail = exc.message
         elif hasattr(exc, "messages"):
             detail = exc.messages
-
         exc = DRFValidationError(detail=detail)
+    elif isinstance(exc, TransitionNotAllowed):
+        detail = str(exc)
+        exc = ParseError(detail=detail)
```

Comment:
A `DRFValidationError` looks more adapted nope ? Or we should create a custom APIException error ?